core/market/src/db/dao/agreement.rs

```diff
@@ -296,6 +346,40 @@ fn update_state(
     Ok(num_updated > 0)
 }
 
+fn update_signature<ColumnSelector>(
+    conn: &ConnType,
+    column: ColumnSelector,
+    agreement: &mut Agreement,
+    signature: String,
+) -> Result<bool, AgreementDaoError>
+where
+    ColumnSelector:
+        diesel::expression_methods::ExpressionMethods + Column<Table = market_agreement>,
+{
+    let signature = Some(signature);
+    let num_updated = diesel::update(market_agreement.find(&agreement.id))
+        .set(column.eq(&signature))
+        .execute(conn)
+        .map_err(|e| AgreementDaoError::DbError(e.into()))?;
+
+    agreement.approved_signature = signature;
```

Comment:
⚠️ bug. You do not know which column to update on the struct, so you can not do it inside this fn.
Unless you select the db again here. which is not most optimal, but acceptable.